src/client/particles.cpp

```diff
@@ -704,17 +702,21 @@ void ParticleManager::clearAll()
 {
 	MutexAutoLock lock(m_spawner_list_lock);
 	MutexAutoLock lock2(m_particle_list_lock);
-	for (auto i = m_particle_spawners.begin(); i != m_particle_spawners.end();) {
-		delete i->second;
-		m_particle_spawners.erase(i++);
+
+	// clear particle spawners
+	for (auto it = m_particle_spawners.begin(); it != m_particle_spawners.end();) {
+		it->second.reset();
+		it = m_particle_spawners.erase(it);
 	}
```

Comment:
Should be done consistently like `m_particles` below, which also looks better.

Reply:
I'm seeing right now, the particlespawner class has a simple default destructor, so we don't even need to iterate.